app/src/test/java/com/kickstarter/viewmodels/ViewPledgeViewModelTest.java

```diff
@@ -293,43 +273,73 @@ public void testShipping_withShippingLocation() {
+    this.shippingLocationTextViewText.assertValues("Sydney, AU");
+    this.shippingAmountTextViewText.assertValues("$5");
+    this.shippingSectionIsGone.assertValues(false);
+  }
 
-    vm.intent(intent(backing));
+  @Test
+  public void testStartMessagesActivity() {
+    final Backing backing = BackingFactory.backing();
+    setUpEnvironment(envWithBacking(backing));
 
-    shippingLocationTextViewTextTest.assertValues("Sydney, AU");
-    shippingAmountTextViewTextTest.assertValues("$5");
-    shippingSectionIsHiddenTest.assertValues(false);
-  }
+    this.vm.intent(new Intent().putExtra(IntentKey.PROJECT, backing.project()));
+    this.vm.inputs.viewMessagesButtonClicked();
 
-  private @NonNull ApiClientType apiClient(final @NonNull Backing backing) {
-    return new MockApiClient() {
-      @Override
-      public @NonNull Observable<Backing> fetchProjectBacking(final @NonNull Project project, final @NonNull User user) {
-        return Observable.just(backing);
-      }
-    };
+    this.startMessagesActivity.assertValues(Pair.create(backing.project(), backing));
   }
 
-  private @NonNull Environment environment(final @NonNull Backing backing) {
-    return environment().toBuilder()
-      .apiClient(apiClient(backing))
-      .currentUser(new MockCurrentUser(backing.backer()))
+  @Test
+  public void testViewMessagesButtonIsGone() {
+    final Backing backing = BackingFactory.backing();
+    final Config config = ConfigFactory.config()
+      .toBuilder()
+      .features(Collections.emptyMap())
       .build();
+
+    final MockCurrentConfig currentConfig = new MockCurrentConfig();
+    currentConfig.config(config);
+
+    setUpEnvironment(envWithBacking(backing).toBuilder().currentConfig(currentConfig).build());
+    this.vm.intent(new Intent().putExtra(IntentKey.PROJECT, backing.project()));
+
+    this.viewMessagesButtonIsGone.assertValues(true);
   }
 
-  private @NonNull Intent intent(final @NonNull Backing backing) {
-    return new Intent().putExtra(IntentKey.PROJECT, backing.project());
+  @Test
+  public void testViewMessagesButtonIsVisible() {
+    final Backing backing = BackingFactory.backing();
+    final Config config = ConfigFactory.config()
+      .toBuilder()
+      .features(Collections.singletonMap(FeatureKey.ANDROID_MESSAGES, true))
+      .build();
+
+    final MockCurrentConfig currentConfig = new MockCurrentConfig();
+    currentConfig.config(config);
+
+    setUpEnvironment(envWithBacking(backing).toBuilder().currentConfig(currentConfig).build());
+    this.vm.intent(new Intent().putExtra(IntentKey.PROJECT, backing.project()));
+
+    this.viewMessagesButtonIsGone.assertValues(false);
   }
 
-  private @NonNull ViewPledgeViewModel vm(final @NonNull Backing backing) {
-    return new ViewPledgeViewModel(environment(backing));
+  /**
+   * Returns an environment with a backing and the backing's current user.
+   */
+  private @NonNull Environment envWithBacking(final @NonNull Backing backing) {
```

Comment:
this is a unique test file in which we pull away a method to create an environment with an api client that has a `backing` and a CurrentUser which is the `backing.backer()` user...decided to condense and keep this logic at the bottom of the file